src/messenger.rs

```diff
@@ -469,6 +469,18 @@ where
 
         Err(SyncVersionsError::NoWorkingVersion)
     }
+
+    pub async fn sasl_handshake(
+        &self,
+        mechanism: &str,
+        auth_bytes: Vec<u8>,
+    ) -> Result<(), RequestError> {
+        let req = SaslHandshakeRequest::new(mechanism);
+        self.request(req).await?;
+        let req = SaslAuthenticateRequest::new(auth_bytes);
+        self.request(req).await?;
+        Ok(())
+    }
 }
```

Comment:
I was really surprised that this "just works" in our CI tests. In short: It doesn't. You have to check the error codes within the messages (`SaslHandshakeResponse::error_code` and `SaslAuthenticateResponse::error_code`) and return an error if the codes are set.

You may wonder how I found that out that quickly: Wireshark:

![image](https://user-images.githubusercontent.com/1529400/165715046-68862754-35cd-4caa-b198-be4546b311ad.png)

Reply:
Ugh, can't believe I missed that! Fixing.